### ISSUE TITLE
Quarkus And XML Routes

### ISSUE THREAD

**Issue by Namphibian** [apache/camel-quarkus#1436]
Hi Team.

Just seeking some guidance.

We have a large number of XML based routes. We would like to convert some of our project over to Quarkus however I am struggling as the lead to find information around how to go about this. We rarely drop down to java as XML can pretty much handle everything.

We tend to use the following structure:
1. beans.xml 
2. camel.xml 
3. route-xxxx.xml

The camel.xml file typically has the following imports:

      <?xml version="1.0" encoding="UTF-8"?>
       <beans xmlns="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd http://camel.apache.org/schema/spring https://camel.apache.org/schema/spring/camel-spring.xsd">
      <import resource="beans.xml"/>
      <import resource="get-work-order-proxy.xml"/>
      <camelContext id="work-order-context" streamCache="true"
                  trace="false" xmlns="http://camel.apache.org/schema/spring"
                  messageHistory="true">
      <routeContextRef ref="get-work-order-proxy-context"/>

When I try the same code in Quarkus I run into a spring name spaces issue. Which is understandable as Quarkus is not spring. So some direct questions.

1. Is there a future for the XML DSL in quarkus camel?
2. Assuming there is a future how do I go about structuring my routes as outlined above.

Appreciate all feedback
N

**Comment by ppalaga** [comment]
The current maintainers of Camel Quarkus are definitely interested in a good XML routes support. However, any future plans depend on the interest and participation of the community.

The set of scenarios we currently test and thus support is rather narrow:

* https://github.com/apache/camel-quarkus/tree/master/examples/timer-log-xml
* https://github.com/apache/camel-quarkus/tree/master/integration-tests/main-xml-io
* https://github.com/apache/camel-quarkus/tree/master/integration-tests/main-xml-jaxb

See also https://camel.apache.org/camel-quarkus/latest/extensions/xml-io.html and https://camel.apache.org/camel-quarkus/latest/extensions/xml-jaxb.html

Please file issues for any missing parts, ideally with small reproducers.

**Comment by davsclaus** [comment]
You have a spring xml file with <beans> - there is no future in that.

You can use Camel XML routes, eg as Peter points out - eg XML files with <routes> (not spring xml <beans> files)

**Comment by gus-maurizio** [comment]
we struggled with the same problem and were able to move most of the routes to the xml-jaxb method. Regarding beans, it is not too difficult to manage them. Thanks to @davsclaus and @lburgazzoli we have good pointers. Private message me for details, specially around AWS credential management (most of our beans were build for spring boot camel around the challenges of proxies, custom endpoints, and the likes).

**Comment by gus-maurizio** [comment]
Now, running a short lived lambda (10 minutes) that starts routes is a different story... :)

**Comment by davsclaus** [comment]
Yeah we wont do a spring xml like for any kind of bean dependency injection - even spring boot is pushing away from xml configuration and into java @Bean annotations.

If you do not want to code at all, then maybe Camel K is more for you.

**Comment by Namphibian** [comment]
Hey Team.

Thanks for the pointers. I understand the beans might be something we need to let go off. Been doing a lot more configuration in SB via yaml and properties as of late. Just trying to align my head around how we can migrate our stuff. I am running 4 AKS clusters in production using Camel primarily. Using Knative and Quarkus together is where I am heading as I would like to leverage Knative eventing.

In short let me try and get one of our applications migrated and maybe put something together for publishing on how to do it.

I might pose a few more questions on here or would you prefer stackoverflow?

**Comment by lburgazzoli** [comment]
@Namphibian stackoverflow is fine but we also have a gitter room https://gitter.im/apache/camel-quarkus

**Comment by Namphibian** [comment]
Thanks mate. I will ask on the gitter as well. Starting to put some more exampled based on our work together will submit some more examples.

**Comment by Namphibian** [comment]
Going to close this and work on some examples of how to migrate. Thanks for the input.
